networks/p2p/server_test.go

```diff
@@ -140,6 +168,58 @@ func TestServerListen(t *testing.T) {
 	}
 }
 
+func TestMultiChannelServerListen(t *testing.T) {
+	// start the test server
+	connected := make(chan *Peer)
+	remid := randomID()
+	config := &Config{ListenAddr: "127.0.0.1:33331", SubListenAddr: []string{"127.0.0.1:33333"}}
+	srv := startTestMultiChannelServer(t, remid, func(p *Peer) {
+		if p.ID() != remid {
+			t.Error("peer func called with wrong node id")
+		}
+		if p == nil {
+			t.Error("peer func called with nil conn")
+		}
+		connected <- p
+	}, config)
+	defer close(connected)
+	defer srv.Stop()
+
+	// dial the test server
+	var defaultConn net.Conn
+
+	for i, address := range srv.GetListenAddress() {
+		conn, err := net.DialTimeout("tcp", address, 5*time.Second)
+		defer conn.Close()
+
+		if i == ConnDefault {
+			defaultConn = conn
+		}
+
+		if err != nil {
+			t.Fatalf("could not dial: %v", err)
+		}
+
+		c := makeconn(conn, randomID())
+		c.doConnTypeHandshake(c.conntype)
+	}
+
+	select {
+	case peer := <-connected:
+		if peer.LocalAddr().String() != defaultConn.RemoteAddr().String() {
+			t.Errorf("peer started with wrong conn: got %v, want %v",
+				peer.LocalAddr(), defaultConn.RemoteAddr())
+		}
+
+		peers := srv.Peers()
+		if !reflect.DeepEqual(peers, []*Peer{peer}) {
+			t.Errorf("Peers mismatch: got %v, want %v", peers, []*Peer{peer})
+		}
+	case <-time.After(5 * time.Second):
+		t.Error("server did not accept within one second")
+	}
```

Comment:
Don't we need check twice  `<-connected` for multi channel?

Reply:
When a peer is added, the peer is received at `connected`. However, only two connections are received, so only one connection is received at `connected`.